utils/s2n_map.c

```diff
@@ -246,3 +246,73 @@ S2N_RESULT s2n_map_free(struct s2n_map *map)
 
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_map_size(struct s2n_map *map, uint32_t *size)
+{
+    RESULT_ENSURE_REF(map);
+    *size = map->size;
+    return S2N_RESULT_OK;
+}
+
+/* Update the internal state so that `current_index` will point to the next value
+ * in the table or will equal `map->capacity` if there are no more elements in
+ * the map. 
+ */
+S2N_RESULT s2n_map_iterator_advance(struct s2n_map_iterator *iter)
+{
+    RESULT_ENSURE_REF(iter);
+    RESULT_ENSURE_REF(iter->map);
+
+    while (++iter->current_index < iter->map->capacity) {
```

Comment:
Talked offline-- "++" like this in a condition can be confusing. And if you remove it, it makes it seem odd that you're using the index becoming too large to track "has_next". I'd suggest a flag instead, which also should avoid the initial iteration in the init method.